agent/src/main/java/com/teamscale/jacoco/agent/Agent.java

```diff
@@ -147,23 +147,30 @@ protected void prepareShutdown() {
 	}
 
 	/**
-	 * Delete a directory from disk if it is empty. This method does nothing if the
-	 * path provided does not exist or point to a file.
+	 * Delete a directory from disk if it is empty. This method does nothing if the path provided does not exist or
+	 * point to a file.
 	 *
-	 * @throws IOException
-	 *             if the deletion of the directory fails
+	 * @throws IOException if the deletion of the directory fails
 	 */
 	private static void deleteDirectoryIfEmpty(Path directory) throws IOException {
-		if (Files.isDirectory(directory) && Files.list(directory).toArray().length == 0) {
-			Files.delete(directory);
+		if (!Files.isDirectory(directory)) {
+			return;
 		}
+
+		try (Stream<Path> stream = Files.list(directory)) {
+			if (stream.count() == 0) {
+				return;
+			}
+		}
+
+		Files.delete(directory);
```

Comment:
This keeps the directory now if it is empty and attempt to delete it if it is not empty, but the code before did the opposite (delete iff empty).

Reply:
whoops, good catch! thanks for the thorough review!